src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -176,19 +183,55 @@
         stats))
     (empty-stats-map 0)))
 
+(s/defn fingerprint-fields-for-db!*
+  "Invokes `fingerprint-fields!` on every table in `database`"
+  ([database :- i/DatabaseInstance
+    tables :- [i/TableInstance]
+    log-progress-fn]
+   (fingerprint-fields-for-db!* database tables log-progress-fn (constantly true)))
+  ;; TODO: Maybe the driver should have a function to tell you if it supports fingerprinting?
```

Comment:
Why wouldn't it? Performance reasons?

Reply:
not sure. this is a carryover from previous. I _think_ its really because google analytics does not support fingerprinting. and rather than a one-off in the fingerprinting code:

```clojure
(if (= :googleanalytics (:engine database))
  (empty-stats-map 0)
  (fingerprint-fields! table))
```

we should just have the driver say this rather than the running code.

Reply:
https://github.com/metabase/metabase/blob/master/src/metabase/sync/analyze/fingerprint.clj#L189 on master

Reply:
I have zero idea if this is still relevant/true. But the intent seems pretty clear

Reply:
Fair. I guess it's due to the fact we can't do `table-rows-sample` for GA. Would probably make sense to generalize the feature flag into something like `:mbql`

Reply:
there's also no point in fingerprinting GA. we should be able to infer the type of each "column" from the api spec here.

Reply:
Fingerprinting also populates the `:fingerprint` field in `Field` model

Reply:
@salsakran sorry but wanted to clarify: are you saying extra work is needed in this PR to infer column information from an api spec or that this is already done and no further work is needed?

Reply:
Google analytics isn't a free form database with tables + columns that can mean anything. There are effectively only metrics and dimensions and those are pretty tightly scoped by the API. Hence we (in general) shouldn't need to do classification. FieldValues we probably still need to do, but arguably we should be able to statically infer most metadata we need.